src/commonMain/kotlin/io/github/detekt/sarif4k/SarifSerializer.kt

```diff
@@ -15,4 +20,54 @@ object SarifSerializer {
     fun toJson(sarif: SarifSchema210): String = json.encodeToString(sarif)
 
     fun fromJson(json: String): SarifSchema210 = Json.decodeFromString(json)
-}
\ No newline at end of file
+}
+
+// Extracted from https://github.com/Kotlin/kotlinx.serialization/issues/296#issuecomment-1859572541
+private fun Any?.toJsonElement(): JsonElement = when (this) {
+    null -> JsonNull
+    is JsonElement -> this
+    is Map<*, *> -> toJsonElement()
+    is Collection<*> -> toJsonElement()
+    is ByteArray -> this.toList().toJsonElement()
+    is CharArray -> this.toList().toJsonElement()
+    is ShortArray -> this.toList().toJsonElement()
+    is IntArray -> this.toList().toJsonElement()
+    is LongArray -> this.toList().toJsonElement()
+    is FloatArray -> this.toList().toJsonElement()
+    is DoubleArray -> this.toList().toJsonElement()
+    is BooleanArray -> this.toList().toJsonElement()
+    is Array<*> -> toJsonElement()
+    is Boolean -> JsonPrimitive(this)
+    is Number -> JsonPrimitive(this)
+    is String -> JsonPrimitive(this)
+    is Enum<*> -> JsonPrimitive(this.toString())
+    else -> error("Can't serialize unknown type: $this")
+}
+
+internal fun Map<*, *>.toJsonElement(): JsonObject {
+    return JsonObject(this.map { (key, value) -> key as String to value.toJsonElement() }.toMap())
```

Comment:
toString instead of `as` to prevent ClassCastException?

Reply:
I think that I prefer to get `ClassCastException`. We give "freedom" here. But you should think that your `Any?` should be parsed to a `json` so the keys should be `String`. I think that it will prevent ugly behaviors.

Reply:
ah, I see where this is used, `is Map<*, *>` cannot be checked for specific key:
![image](https://github.com/detekt/sarif4k/assets/2906988/72345655-2d72-48f7-8b5a-20fc12698be2)
so we have to do it runtime

Reply:
one thing we might want to do:

```suggestion
    return JsonObject(this.map { (key, value) -> (key as? String ?: error("$key is not allowed as JSON key, please use a String!")) to value.toJsonElement() }.toMap())
```